src/charm.py

```diff
@@ -173,13 +174,31 @@ def _on_config_changed(self, _: ops.HookEvent) -> None:
     def _on_synapse_pebble_ready(self, _: ops.HookEvent) -> None:
         """Handle synapse pebble ready event."""
         self.change_config()
+        # Since Synapse is ready, the charm can create the admin access token
+        container = self.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for Synapse pebble")
+            return
+        time.sleep(3)
+        admin_user = synapse.create_admin_user(container)
+        if not admin_user:
+            logger.error("Failed to create admin user.")
+            return
+        self._set_admin_access_token(admin_user.access_token)
+        self._charm_state.synapse_config.admin_access_token = admin_user.access_token
+        # Mjolnir is a moderation tool for Matrix.
+        # See https://github.com/matrix-org/mjolnir/ for more details about it.
+        if self._charm_state.synapse_config.enable_mjolnir:
+            self._mjolnir = Mjolnir(self, charm_state=self._charm_state)
```

Comment:
I believe at this point there could be two Mjolnir instances, each one of them registered to the collect_unit_status event (one created in `Charm.__init__` and one here).

Reply:
Removed ;)
Yesterday, I had a completely different idea to remove the dependency between user creation and events. I'll mark this and the following comments as resolved because it has changed a lot by adding a "Token Service". Reviews are welcome! :)